File: benchmark/comparison/results/haiku_ts/06.ts
function runningSum(lst: number[]): number[] {
  const result: number[] = [];
  let sum = 0;
  for (const num of lst) {
    sum += num;
    result.push(sum);
  }
  return result;
}

function formatList(lst: number[]): string {
  return "[" + lst.join(", ") + "]";
}

// Test cases
console.log(formatList(runningSum([1, 2, 3, 4])));
console.log(formatList(runningSum([])));
console.log(formatList(runningSum([5])));
console.log(formatList(runningSum([-1, 2, 3])));

File: benchmark/comparison/results/haiku_py/06.py
def running_sum(lst):
    result = []
    total = 0
    for num in lst:
        total += num
        result.append(total)
    return result

# Test
print(running_sum([1, 2, 3, 4]))
print(running_sum([]))
print(running_sum([5]))
print(running_sum([-1, 2, 3]))
